php-classes/Slate/Courses/Section.php

```diff
@@ -411,6 +419,68 @@ public function getCohorts()
         }
     }
 
+    public function findBlogTags()
+    {
+        return TagItem::getTagsSummary([
+            'Class' => BlogPost::class,
+            'classConditions' => [
+                'ContextClass' => static::getStaticRootClass(),
+                'ContextID' => $this->ID
+            ]
+        ]);
+    }
+
+    public function findBlogPosts($conditions, $limit, $offset, $tag)
+    {
+      $options = [
+        'limit' => $limit,
+        'offset' => $offset,
+        'calcFoundRows' => 'yes',
+        'conditions' => $conditions
+      ];
+
+      if ($tag!=null) {
+
+          $tagIDsQuery = 'SELECT ContextID FROM `tag_items` WHERE (`ContextClass` = "%s") AND (`TagID` = %u)';
+          $tagIDsConditions = [
+              'ContextClass' => DB::escape(BlogPost::getStaticRootClass()),
+              'TagID' => $tag->ID
+          ];
+
+          $tagItemIDs = DB::allValues('ContextID', $tagIDsQuery, $tagIDsConditions);
+
+          $options = array_merge_recursive($options, [
+              'conditions' => [
+                  'ID' => [
+                      'operator' => 'IN',
+                      'values' => count($tagItemIDs) ? $tagItemIDs : '0'
+                  ]
+              ]
+          ]);
+      }
+
+      return BlogPost::getAllPublishedByContextObject($this, $options);
+    }
+
+    public function findLatestTeacherPost()
+    {
+      $sectionTeacherIds = array_map(function($Teacher) {
+          return $Teacher->ID;
+      }, $this->ActiveTeachers);
+
+      $latestTeacherPost = \Emergence\CMS\BlogPost::getAllPublishedByContextObject($this, array_merge_recursive([
+          'conditions' => [
+              'AuthorID' => [
+                  'operator' => 'IN',
+                  'values' => $sectionTeacherIds
+              ]
+          ],
+          'limit' => 1
+      ]));
+
+      return $latestTeacherPost[0];
```

Comment:
to avoid undefined index error if there are no results:

```suggestion
      return count($latestTeacherPost) ? $latestTeacherPost[0] : null;
```